### PR TITLE
Cleanup Dockerfiles

### PR DESCRIPTION
**Pull request by marstr**
- Updating the name of the Docker organization the images should be tagged with.
- Adding `npm install` step as part of container build
- Adding `proxy-docker` and `docker` make targets.

### CODE REVIEW

**Comment by marstr** [comment]
Rebased my changes on master to stay up-to-date, same changes though.

**Comment by michalpristas** [comment]
@arschles want to review as a local Docker guru?

**Comment by marstr** [comment]
Looks like CI is failing for reasons unrelated to this PR. :/

**Comment by robjloranger** [comment]
I'm not sure why this one keeps failing with a timeout 🤔

**Comment by robjloranger** [comment]
Looks all good except the conflict in `Makefile` @marstr if you can update a resolve we can merge

File: cmd/olympus/Dockerfile [hunk 1]
```diff
@@ -8,9 +8,8 @@ FROM gobuffalo/buffalo:v0.11.0 as builder
 RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 ADD . .
-WORKDIR $GOPATH/src/github.com/gomods/athens/cmd/olympus
 
-RUN buffalo build -s -o /bin/app
+RUN cd cmd/olympus && npm install && buffalo build -s -o /bin/app
```
Comment (arschles) [inline cmd/olympus/Dockerfile]: @marstr I think you can do a `yarn install` above this line, similar to what the proxy does in https://github.com/gomods/athens/blob/master/cmd/proxy/Dockerfile#L11

What do you think?

File: cmd/proxy/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ RUN yarn install --no-progress
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
 ADD . .
-RUN cd cmd/proxy && buffalo build -s -o /bin/app
+RUN cd cmd/proxy && npm install && buffalo build -s -o /bin/app
```
Comment (arschles) [inline cmd/proxy/Dockerfile]: @marstr do you need the `npm install` here? I noticed that there is a `yarn install` above

Comment (marstr) [reply cmd/proxy/Dockerfile]: Derp, sorry about that. Fixed.

File: cmd/olympus/Dockerfile (hunk 1 above)
Comment (marstr) [reply cmd/olympus/Dockerfile]: Fixed in 484fd0e.

**Comment by arschles** [comment]
Ref https://github.com/gomods/athens/issues/81

**Pull request merged**